react/src/components/AssetDetail/AssetDetail.tsx

```diff
@@ -0,0 +1,172 @@
+import React, { Suspense, useState, useEffect } from 'react';
+import _ from 'lodash';
+import { useAppConfiguration, useFeatureSelection } from '@hazmapper/hooks';
+import { useLocation, useNavigate } from 'react-router-dom';
+import { Asset, FeatureTypeNullable } from '@hazmapper/types';
+import { FeatureIcon } from '@hazmapper/components/FeatureIcon';
+import { Button, LoadingSpinner, SectionMessage } from '@tacc/core-components';
+import styles from './AssetDetail.module.css';
+import { faCameraRetro, faPhotoFilm } from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+
+type AssetModalProps = {
+  isOpen: boolean;
+  projectId: number;
+  isPublicView: boolean;
+};
+
+const AssetDetail: React.FC<AssetModalProps> = ({
+  isOpen,
+  projectId,
+  isPublicView,
+}) => {
+  const { selectedFeature, selectedFeatureId, setSelectedFeatureId } =
+    useFeatureSelection();
+
+  const navigate = useNavigate();
+  const location = useLocation();
+
+  const closePanel = () => {
+    const params = new URLSearchParams(location.search);
+    params.delete('selectedFeature');
+    navigate(`${location.pathname}?${params.toString()}`);
```

Comment:
could use `setSelectedFeatureId(selectedFeatureId)` instead of this block.

should we rename `setSelectedFeature` to `toggleSelectedFeature` as it toggle it if already selected?

Reply:
Would this affect the stack of queries that the hook is sorting through in useCurrentFeature hook if it's not deleted? If not, I can use the solution you proposed here: [https://github.com/TACC-Cloud/hazmapper/pull/292#discussion_r1884610627](https://github.com/TACC-Cloud/hazmapper/pull/292#discussion_r1884610627)